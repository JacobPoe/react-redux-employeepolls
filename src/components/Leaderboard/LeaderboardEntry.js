import { Card } from 'react-bootstrap';
import { Image } from 'react-bootstrap';

const LeaderboardEntry = ({ user }) => {
  return (
    <>
      <Card className="leaderboard-entry">
        <Card.Body>
          <Image src={user.avatarURL} thumbnail />
          <Card.Title>{user.name}</Card.Title>
          <Card.Subtitle>@{user.id}</Card.Subtitle>
          <hr />
          <Card.Text># of Questions Asked: {user.questions.length}</Card.Text>
          <Card.Text>
            # of Questions Answered: {Object.keys(user.answers).length}
          </Card.Text>
          <hr />
          <Card.Text>
            Total Votes Cast:{' '}
            {user.questions.length + Object.keys(user.answers).length}
          </Card.Text>
        </Card.Body>
      </Card>
    </>
  );
};

export default LeaderboardEntry;
