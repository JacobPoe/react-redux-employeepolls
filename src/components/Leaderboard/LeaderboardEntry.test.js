import * as React from 'react';
import { render } from '@testing-library/react';

import LeaderboardEntry from "./LeaderboardEntry"

describe('LeaderboardEntry', () => {
  const user = {
    "id": "mtsamis",
    "password": "xyz123",
    "name": "Mike Tsamis",
    "avatarURL": "https://www.bungie.net/common/destiny2_content/icons/23e3d07bead2936a80612419f3ee9f94.jpg",
    "answers": {
        "xj352vofupe1dqz9emx13r": "optionOne",
        "vthrdm985a262al8qx3do": "optionTwo",
        "6ni6ok3ym7mf1p33lnez": "optionOne"
    },
    "questions": [
        "6ni6ok3ym7mf1p33lnez",
        "xj352vofupe1dqz9emx13r"
    ]
  }

  it('should properly render the leaderboard card for user "mtsamis"', () => {
    const component = render(<LeaderboardEntry user={user} />)
    expect(component).toMatchSnapshot();
  })
})