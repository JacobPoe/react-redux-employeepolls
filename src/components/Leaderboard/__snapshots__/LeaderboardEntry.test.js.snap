// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LeaderboardEntry should properly render the leaderboard card for user "mtsamis" 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="leaderboard-entry card"
      >
        <div
          class="card-body"
        >
          <img
            class="img-thumbnail"
            src="https://www.bungie.net/common/destiny2_content/icons/23e3d07bead2936a80612419f3ee9f94.jpg"
          />
          <div
            class="card-title h5"
          >
            Mike Tsamis
          </div>
          <div
            class="card-subtitle h6"
          >
            @
            mtsamis
          </div>
          <hr />
          <p
            class="card-text"
          >
            # of Questions Asked: 
            2
          </p>
          <p
            class="card-text"
          >
            # of Questions Answered: 
            3
          </p>
          <hr />
          <p
            class="card-text"
          >
            Total Votes Cast:
             
            5
          </p>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="leaderboard-entry card"
    >
      <div
        class="card-body"
      >
        <img
          class="img-thumbnail"
          src="https://www.bungie.net/common/destiny2_content/icons/23e3d07bead2936a80612419f3ee9f94.jpg"
        />
        <div
          class="card-title h5"
        >
          Mike Tsamis
        </div>
        <div
          class="card-subtitle h6"
        >
          @
          mtsamis
        </div>
        <hr />
        <p
          class="card-text"
        >
          # of Questions Asked: 
          2
        </p>
        <p
          class="card-text"
        >
          # of Questions Answered: 
          3
        </p>
        <hr />
        <p
          class="card-text"
        >
          Total Votes Cast:
           
          5
        </p>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
